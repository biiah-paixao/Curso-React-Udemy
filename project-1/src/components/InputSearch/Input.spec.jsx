import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { InputSearch } from '.';

describe('<InputSearch />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<InputSearch type="search" searchValue="testando" handleChange={fn} />);

    const inputSearch = screen.getByPlaceholderText(/type your search/i);
    expect(inputSearch.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<InputSearch type="search" handleChange={fn} searchValue="um valor" />);

    const inputSearch = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';

    userEvent.type(inputSearch, value);

    expect(inputSearch.value).toBe('um valor');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<InputSearch handleChange={fn} searchValue="testando" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
