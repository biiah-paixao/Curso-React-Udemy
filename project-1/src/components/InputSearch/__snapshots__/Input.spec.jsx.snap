// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch /> should match snapshot 1`] = `
<input
  class="search-input"
  placeholder="Type your search"
  type="search"
  value="testando"
/>
`;
